File: src/neatwork/core/SimuJava/ressource/JSONToExcelConverter.py
import json
import openpyxl

def convert_json_to_excel(json_file, excel_file):
    # Ouvrir le fichier JSON
    with open(json_file, 'r') as f:
        data = json.load(f)
    
    # Créer un nouveau classeur Excel
    workbook = openpyxl.Workbook()
    sheet = workbook.active
    row = 1
    
    # Parcourir les clés du JSON
    for _, (key, value) in enumerate(data.items(), start=1):
        # Écrire la clé dans la première colonne
        sheet.cell(row=row, column=1, value=key)

        row += 1
        
        # Si la valeur est une matrice, écrire chaque ligne sur une ligne différente
        if key == "S":
            for row_index, row_values in enumerate(value):
                for col_index, val in enumerate(row_values, start=1):
                    sheet.cell(row=row, column=col_index, value=val)
                row += 1
        else:
            # Si la valeur est une liste, écrire chaque élément sur la ligne suivante
            if isinstance(value, list):
                for col_index, item in enumerate(value, start=1):
                    sheet.cell(row=row, column=col_index, value=item)
            else:
                # Si la valeur est un scalaire, écrire dans la ligne suivante
                sheet.cell(row=row, column=1, value=value)
        
        # Passe à la ligne suivante
        row += 2
    
    # Sauvegarder le classeur Excel
    workbook.save(excel_file)
    print("Conversion terminée avec succès.")

# Utilisation de la fonction pour convertir le fichier JSON en Excel
convert_json_to_excel("SimuJava/src/dat.json", "SimuJava/src/dat.xlsx")

File: src/neatwork/core/SimuJava/ressource/JSONToExcelConverter.java
package neatwork.core.SimuJava.ressource;
import org.apache.poi.ss.usermodel.*;
import org.apache.poi.xssf.usermodel.XSSFWorkbook;
import org.json.JSONArray;
import org.json.JSONObject;

import java.io.FileReader;
import java.io.FileOutputStream;
import java.io.IOException;

public class JSONToExcelConverter {

    public static void main(String[] args) {
        convertJSONToExcel("SimuJava/src/dat.json", "SimuJava/src/output.xlsx");
    }

    public static void convertJSONToExcel(String jsonFilePath, String excelFilePath) {
        try {
            // Lire le fichier JSON
            FileReader reader = new FileReader(jsonFilePath);
            JSONObject jsonObject = new JSONObject(reader);

            // Créer un nouveau fichier Excel
            Workbook workbook = new XSSFWorkbook();
            Sheet sheet = workbook.createSheet("Data");

            int rowNum = 0;

            // Parcourir les clés du JSON
            for (String key : jsonObject.keySet()) {
                // Écrire la clé dans la première colonne de chaque ligne
                Row row = sheet.createRow(rowNum++);
                Cell cell = row.createCell(0);
                cell.setCellValue(key);

                // Récupérer les valeurs correspondantes
                Object value = jsonObject.get(key);

                // Si la valeur est un tableau, écrire chaque élément dans une colonne séparée
                if (value instanceof JSONArray) {
                    JSONArray jsonArray = (JSONArray) value;
                    for (int i = 0; i < jsonArray.length(); i++) {
                        Row dataRow = sheet.createRow(rowNum++);
                        Cell dataCell = dataRow.createCell(0);
                        dataCell.setCellValue(jsonArray.getDouble(i));
                    }
                } else {
                    // Si la valeur est un scalaire, écrire dans la deuxième colonne
                    Cell valueCell = row.createCell(1);
                    if (value instanceof Double) {
                        valueCell.setCellValue((Double) value);
                    } else if (value instanceof Integer) {
                        valueCell.setCellValue((Integer) value);
                    } else if (value instanceof String) {
                        valueCell.setCellValue((String) value);
                    }
                }

                // Sauter une ligne après chaque clé
                rowNum++;
            }

            // Écrire le contenu dans un fichier Excel
            try (FileOutputStream fileOut = new FileOutputStream(excelFilePath)) {
                workbook.write(fileOut);
            }

            // Fermer le workbook
            workbook.close();

            System.out.println("Conversion terminée avec succès.");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
